tests/ctst/features/resource-policies/AssumeRole.feature

```diff
@@ -0,0 +1,1283 @@
+Feature: S3 Bucket Policies Authorization flow for IAM Assume Roles
+    This feature allows you to create and attach bucket policies to S3 buckets.
+    IAM Users should have the permissions to perform the actions that they are granted in their bucket policies
+    based on the other permissions they also have from their role.
+    This test suite is not meant to be human-readable, but brings confidence in our Authz flow for all supported
+    S3 actions.
+
+    @2.6.0
+    @PreMerge
+    @BucketPolicies
+    @BP-ASSUME_ROLE_USER
+    Scenario Outline: ASSUME ROLE: <iamPolicyExists> <iamPolicyApplies> <iamPolicyEffect> IAM Policy and <bucketPolicyExists> <bucketPolicyApplies> <bucketPolicyEffect> S3 Bucket Policy
+        Given an action "<action>"
+        And an existing bucket prepared for the action
```

Comment:
Do you think we can generalize the preparation of buckets for actions for all tests?

Reply:
Not sure, here we infer the config from the API, while other APIs might need to configure it with specific data; that should stay in the scenario definition. Here it's mostly because we have too many APIs to test for authorization